src/walk.rs

```diff
@@ -170,12 +170,17 @@ fn spawn_receiver(
 
     let show_filesystem_errors = config.show_filesystem_errors;
     let threads = config.threads;
-
+    // This will be used to check if output should be buffered when only running a single thread
+    let is_multithread: bool = if threads > 1 {
```

Comment:
Added an 'is_multithread' bool that will be passed down to execute_command

Reply:
``` rs
if threads > 1 {
        true
    } else {
        false
    }
```
could be just `threads > 1`.

Also, maybe we should rename the variable and the parameter to `enable_output_buffering`? On the callee side, I think this would be much clearer. And here, on the caller side, we could add a comment like:
``` rust
// We only buffer the output if we are using multiple threads. Otherwise, parallel execution
// can lead to interleaved output:
let enable_output_buffering = threads > 1;
```